Source/missiles.cpp

```diff
@@ -1898,29 +1898,23 @@ void AddMagmaball(Missile &missile, Point dst, Direction /*midir*/)
 
 void AddTeleport(Missile &missile, Point dst, Direction /*midir*/)
 {
-	missile._miDelFlag = true;
-
 	for (int i = 0; i < 6; i++) {
 		int k = CrawlNum[i];
 		int ck = k + 2;
 		for (auto j = static_cast<uint8_t>(CrawlTable[k]); j > 0; j--, ck += 2) {
 			Point target = dst + Displacement { CrawlTable[ck - 1], CrawlTable[ck] };
-			if (!InDungeonBounds(target) || IsTileSolid(target))
+			if (!PosOkPlayer(Players[missile._misource], target))
 				continue;
-			if (dMonster[target.x][target.y] == 0 && dObject[target.x][target.y] == 0 && dPlayer[target.x][target.y] == 0) {
-				missile.position.tile = target;
-				missile.position.start = target;
-				missile._miDelFlag = false;
-				i = 6;

```

Comment:
OMG this assignment is insane..... am I understanding this right? Was this a hacky way of "breaking out of the external loop as well"?

Reply:
yep, it's confused me like 50 times by now.